README.md

```diff
@@ -6,15 +6,41 @@ directly with the file transfer service.
 
 [![Build Status](https://travis-ci.org/hmrc/upscan-initiate.svg)](https://travis-ci.org/hmrc/upscan-initiate) [ ![Download](https://api.bintray.com/packages/hmrc/releases/upscan-initiate/images/download.svg) ](https://bintray.com/hmrc/releases/upscan-initiate/_latestVersion)
 
-# Usage
-
-The microservice provides one endpoint that allows to request upload of single file.
-To initiate the upload it's needed to make POST request to the '/upscan/initiate' endpoint. The request
-should contain details about expected upload, which include additional metadata, additional constraints
+# Upscan user manual
+
+## Introduction
+
+Upscan service allows consuming services to orchestrate the process of uploading files by the end users. Upscan provides
+temporary storage of uploaded file, ensures that the file isn't harmful (doesn't contain viruses) and fulfills
+some predefined preconditions (like maximum size).
+The process of uploading and verifying of uploaded file is performed asynchronously without involvement of consuming service.
+
+## File upload workflow
+
+* Consuming service requests upload of a single file. It makes a call to `/upscan/initiate` endpoint with details of
+requested upload, including size constraints and callback URL 
+* Upscan service replies with template of the HTTP POST form that should be used to upload the file and with unique reference
+of the upload that can be later used by the consumer service to identify the file
+* Consuming service returns the form to the user, the form is being displayed in users browser.
+* User uploads the file using his web browser.
+* In the background upscan service performs all necessary check of the file
+* If file is safe, upscan service notifies consuming service, using previously provided callback URL. The callback body contains
+an URL allowing to fetch uploaded file
+* Consuming service downloads the file using provided URL. The link is valid for specified time (TODO - not specified yet)
+* After specified time the file is automatically removed
```

Comment:
Automatically removed from the stored location. The service does NOT keep the file indefinitely.